jupyterhub/auth.py

```diff
@@ -1105,7 +1124,10 @@ def authenticate(self, handler, data):
         username = data['username']
         try:
             pamela.authenticate(
-                username, data['password'], service=self.service, encoding=self.encoding
+                username,
+                (data['password'], data['otp']),
```

Comment:
data['otp'] should only be expected when `self.request_otp`, I think. Alternately, it can be added if present in `data`, but shouldn't be assumed present.